lib/validateInput.js

```diff
@@ -7,26 +7,72 @@ module.exports = function(inputType) {
 	var blendingModeValues = ['multiply', 'screen', 'normal', 'overlay', 'darken',
 		'lighten', 'lighter', 'color-dodge', 'color-burn', 'hard-light', 'soft-light',
 		'difference', 'exclusion', 'hue', 'saturation', 'color', 'luminosity'];
+	var directionValues = ['diagonal', 'left-right', 'top-bottom', 'radial', 'custom'];
 
 	switch(inputType) {
 		case 'image':
 			// Validate image.position
 			if ((!Array.isArray(this.image.position) || this.image.position.length !== 2) ||
 				xPositionValues.indexOf(this.image.position[0]) === -1 ||
 				yPositionValues.indexOf(this.image.position[1]) === -1
-			) {this.triggerError('image.position')}
+			) { this.triggerError('image.position') }
 			// Validate image.stretchMode
 			if (this.image.stretchMode) {
 				if ((!Array.isArray(this.image.stretchMode) || this.image.stretchMode.length !== 2) ||
 					stretchModeValues.indexOf(this.image.stretchMode[0]) === -1 ||
 					stretchModeValues.indexOf(this.image.stretchMode[1]) === -1
-				) {this.triggerError('image.stretchMode')}
+				) { this.triggerError('image.stretchMode') }
 			}
 			break;
 		case 'blendingMode':
 			if (blendingModeValues.indexOf(this.image.blendingMode) === -1) {
 				this.clear();
 				this.triggerError('blendingMode');
 			}
+			break;
+		case 'direction':
+			if (directionValues.indexOf(this.direction) === -1) {
+				this.triggerError('direction');
+			} else {
+				if (this.direction === 'custom') {
+					if (!areDefinedAndPixelsOrPercentage([
+						this.customDirection.x0,
+						this.customDirection.x1,
+						this.customDirection.y0,
+						this.customDirection.y1
+					])) {
+						this.triggerError('customDirection');
+					}
+				}
+			}
+			break;
 	}
 };
+
+function areDefinedAndPixelsOrPercentage(array) {
```

Comment:
Since you instantiate the variable `unit` outside of the `while` scope if the first value of the array has a `unit` but the second one don't the `!unit` condition will be true.
Should keep:
```
var unit;
if (value.indexOf('px') !== -1) unit = 'px';
if (value.indexOf('%') !== -1) unit = '%';
```

Reply:
Ok, you can instantiate the 2 vars in the while loop.